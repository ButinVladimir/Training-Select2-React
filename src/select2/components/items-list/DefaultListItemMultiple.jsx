import React from 'react';
import PropTypes from 'prop-types';

export default function DefaultListItemMultiple({
  value,
  name,
  checked,
  onChange,
}) {
  const id = `select2-${value}`;

  return (
    <li className={checked ? 'checked' : ''}>
      <label htmlFor={id}>
        <input id={id} type="checkbox" checked={checked} onChange={() => onChange(value, name)} />
        {name}
      </label>
    </li>
  );
}

DefaultListItemMultiple.propTypes = {
  value: PropTypes.any.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};
