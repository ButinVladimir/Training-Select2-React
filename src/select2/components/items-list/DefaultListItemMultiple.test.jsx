import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DefaultListItemMultiple from './DefaultListItemMultiple';

describe('DefaultListItemMultiple', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  test('renders without crashing unchecked', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultListItemMultiple
      checked={false}
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('renders without crashing checked', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultListItemMultiple
      checked
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('fires onChange properly', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<DefaultListItemMultiple
      checked={false}
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const input = wrapper.find('input');
    input.simulate('change', 'value', 'name');
    expect(mockFn).toHaveBeenCalledWith('value', 'name');
  });
});
