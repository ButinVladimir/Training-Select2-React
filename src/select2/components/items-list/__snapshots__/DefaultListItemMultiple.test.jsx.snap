// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultListItemMultiple renders without crashing checked 1`] = `
<li
  className="checked"
>
  <label
    htmlFor="select2-value"
  >
    <input
      checked={true}
      id="select2-value"
      onChange={[Function]}
      type="checkbox"
    />
    name
  </label>
</li>
`;

exports[`DefaultListItemMultiple renders without crashing unchecked 1`] = `
<li
  className=""
>
  <label
    htmlFor="select2-value"
  >
    <input
      checked={false}
      id="select2-value"
      onChange={[Function]}
      type="checkbox"
    />
    name
  </label>
</li>
`;
